src/main/java/frc/robot/subsystems/Grabber.java

```diff
@@ -0,0 +1,123 @@
+
+
+public class Grabber extends SubsystemBase {
+
+  ArmStageOne armStageOne; //refactor this to armStageOneSubsystem >:(
+  ArmStageTwo armStageTwo;
+  EndEffectorSubsystem endEffectorSubsystem;
+  /** Creates a new Grabber. */
+  public Grabber(ArmStageOne armStageOne, ArmStageTwo armStageTwo, EndEffectorSubsystem endEffectorSubsystem) {
+    this.armStageOne = armStageOne;
+    this.armStageTwo = armStageTwo;
+    this.endEffectorSubsystem = endEffectorSubsystem;
+  }
+  public enum ArmModes {INTAKING_CUBE, INTAKING_CONE_UPRIGHT, INTAKING_CONE_FALLEN, HOLDING, PLACING_CONE_LVL1, PLACING_CONE_LVL2, PLACING_CONE_LVL3,PLACING_CUBE_LVL1,PLACING_CUBE_LVL2,PLACING_CUBE_LVL3}
+  public enum EFModes {INTAKING_CONE, INTAKING_CUBE, HOLDING_CUBE, HOLDING_CONE, PLACING_CUBE, PLACING_CONE, STOPPED}
+
+  public void setArmPosition(ArmModes armMode){
+    double[] armPositions = getArmPositions(armMode);
+    if(armPositions.length>=2)
+      setArmPosition(armPositions);
+    this.armMode = armMode;
+  }
+
+  public void setEndEffectorMode(EFModes effectorMode){
+    switch(effectorMode){
+      case INTAKING_CUBE: endEffectorSubsystem.intakeCube(); break;
+      case INTAKING_CONE: endEffectorSubsystem.intakeCone(); break;
+      case HOLDING_CONE: endEffectorSubsystem.holdCone(); break;
+      case HOLDING_CUBE: endEffectorSubsystem.holdCube(); break;
+      case PLACING_CONE: endEffectorSubsystem.placeCone(); break;
+      case PLACING_CUBE: endEffectorSubsystem.placeCube(); break;
+      case STOPPED: endEffectorSubsystem.stopMotors(); break;
+    }
+    efMode = effectorMode;
+  }
+
+  public double[] getArmPositions(ArmModes armMode){
+    switch (armMode){
+      case INTAKING_CUBE: return ArmConstants.intakingCubesArmPos;
+      case INTAKING_CONE_UPRIGHT: return ArmConstants.intakingConesUprightArmPos;
+      case INTAKING_CONE_FALLEN: return ArmConstants.intakingConesFallenArmPos;
+      case HOLDING: return ArmConstants.holdingArmPos;
+
+      case PLACING_CONE_LVL1: return ArmConstants.placingConeArmPosOne;
+      case PLACING_CONE_LVL2: return ArmConstants.placingConeArmPosTwo;
+      case PLACING_CONE_LVL3: return ArmConstants.placingConeArmPosThree;
+
+      case PLACING_CUBE_LVL1: return ArmConstants.placingCubeArmPosOne;
+      case PLACING_CUBE_LVL2: return ArmConstants.placingCubeArmPosTwo;
+      case PLACING_CUBE_LVL3: return ArmConstants.placingCubeArmPosThree;
+    }
+    return new double[]{};
+  }
+
+  private EFModes efMode = EFModes.STOPPED;
+  private ArmModes armMode = ArmModes.HOLDING;
+  private EFModes desiredEFMode = EFModes.STOPPED;
+  private ArmModes desiredArmMode = ArmModes.HOLDING;
+
+  public void setDesiredEFMode(EFModes desiredEFMode){
+    this.desiredEFMode = desiredEFMode;
+  }
+  public void setDesiredArmMode(ArmModes desiredArmMode){
+    this.desiredArmMode = desiredArmMode;
+  }
+  public EFModes getDesiredEFMode(){
+    return desiredEFMode;
+  }
+  public ArmModes getDesiredArmMode(){
```

Comment:
when is a desired EF/Arm mode different from the actual one?

Reply:
The desired EF will be different when the arm isn't yet where it should be. Right now, the desired and actual Arm states should always match.